### PR TITLE
Add Email validation

### PR DESCRIPTION
**Pull request by miaulalala**
Add some mail validation for email addresses in the CalDAV backend

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

File: lib/public/Authentication/LoginCredentials/ICredentials.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ public function getLoginName();
 	 *
 	 * @since 12
 	 *
-	 * @return string|null
```
Comment (tcitworld) [inline lib/public/Authentication/LoginCredentials/ICredentials.php:56]: Looks unrelated.

**Review by tcitworld — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
Performing the fixup

**Comment by ChristophWurst** [comment]
S3 tests timed out -> unrelated.
